### PR TITLE
Add support generic http server handler.

### PR DESCRIPTION
**Pull request by rghetia**

### CODE REVIEW

**Comment by songy23** [comment]
Please add this and https://github.com/census-instrumentation/opencensus-java/pull/1558 to CHANGELOG.md.

File: contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpServerHandler.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import org.checkerframework.checker.nullness.qual.NonNull;
+*/
+
+/**
+ * This helper class provides routine methods to instrument HTTP servers.
+ *
+ * @param <Q> the HTTP request entity.
+ * @param <P> the HTTP response entity.
+ * @param <C> the type of the carrier.
+ * @since 0.17
```
[the remark this answers is not in the document]
Comment (rghetia) [reply contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpServerHandler.java]: done

File: contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpServerHandler.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    SpanBuilder spanBuilder = null;
+    String spanName = getSpanName(request, extractor);
+    // de-serialize the context
+    SpanContext spanContext = null;
+    try {
+      spanContext = textFormat.extract(carrier, getter);
+      if (publicEndpoint) {
+        spanBuilder = tracer.spanBuilder(spanName);
+      } else {
+        spanBuilder = tracer.spanBuilderWithRemoteParent(spanName, spanContext);
```
[the remark this answers is not in the document]
Comment (rghetia) [reply contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpServerHandler.java]: done.

**Comment by rghetia** [comment]
> Please add this and #1558 to CHANGELOG.md.

sure. I'll open a new PR for changelog.

**Pull request merged**